assets/app/view/game/buy_trains.rb

```diff
@@ -127,6 +129,20 @@ def render
           ])
         end
 
+        @slot_checkboxes = {}
+        if step.respond_to?(:slot_view) && (view = step.slot_view(@corporation))
+          children << send("render_#{view}")
+        end
+
+        scrappable_trains = []
+        scrappable_trains = step.scrappable_trains(@corporation) if step.respond_to?(:scrappable_trains)
+        unless scrappable_trains.empty?
+          children << h(:h3, h3_props, 'Trains to Scrap')
+          children << h(:div, div_props, [
```

Comment:
scrap_trains(scrappable_trains)

no need to make an array and then explode an array...

Reply:
Ok